Patient profile modifications

KasiSribal opened the pull request:
Patient Profile Modified - Added Slicers of Identifier Element

enjoysparkling commented:
Move StructureDefinition-profile-patient.json into the `input/resources` folder.  This will make it easier for me to compare the current file with the changes you have here.

enjoysparkling commented:
Accidentally closed it. Make those changes and let me know so I can review.

enjoysparkling commented:
@KasiSribal - I merged the latest 'main' branch changes into this one and moved the Patient profile into the appropriate directory to get this moving.  I will let you handle subsequent changes.

enjoysparkling commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -44,6 +44,216 @@
+      {
+        "id": "Patient.identifier:ProviderClientId.system",
+        "path": "Patient.identifier.system",
+        "fixedUri": "PROVIDER SYSTEM URL"
```
Do not hard code this value.  The developers will have to provide this when they input a ProviderClientID
Add a note in the "Definition" field:  "This needs to be the URL that is specific to the Provider's domain; e.g. http://example.org"

KasiSribal replied:
Removed System -->Fixed value and added definition as mentioned
![image](https://github.com/enjoysparkling/signal-mso-fhir-profiles/assets/136109806/5d0ae0c5-4cf3-4a32-bb7d-37f20e84d0d7)

enjoysparkling replied:
I do not see the Definition on Patient.identifier:ProviderClientId.system.  Would you double-check that it saved properly?

enjoysparkling commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -44,6 +44,216 @@
+        "max": "1"
+      },
+      {
+        "id": "Patient.identifier:ClientId.type.coding.system",
```
Patient.identifier:ClientId represents Signal's Beacon generated number, correct?   I don't think it fits in with anything in the **type.coding.system**, so I suggest removing all entries for type.coding--leave it blank for this one.

KasiSribal replied:
Done

enjoysparkling replied:
@KasiSribal - Sorry to reneg on this.  I was reading through https://www.hl7.org/fhir/valueset-identifier-type.html, and I do think we should call it a "MR" code.  Would you add back the applicable values? i.e.
type.coding.system = https://www.hl7.org/fhir/valueset-identifier-type.html
type.coding.code = MR
type.coding.display = Medical record number

enjoysparkling commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -44,6 +44,216 @@
+        "path": "Patient.identifier",
+        "sliceName": "ClientId",
+        "short": "Client identifier for this patient",
+        "definition": "Client identifier for this patient.",
```
label the definition as "Signal's Beacon legacy identifier for client."

KasiSribal replied:
changed the definition

enjoysparkling commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -44,6 +44,216 @@
+        "fixedUri": "PROVIDER SYSTEM URL"
+      },
+      {
+        "id": "Patient.identifier:ClientId",
```
Since this represents the legacy Signal Beacon identifier, assign the `Patient.identifier:ClientID.system" as `https://signalbhn.org/fhir/sid/beacon-legacy`

KasiSribal replied:
system value changed

enjoysparkling commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -44,6 +44,216 @@
+        "fixedString": "ProviderID"
+      },
+      {
+        "id": "Patient.identifier:DACODSId",
```
I don't think there is a need for Patient.identifier:DACODSId in the new system.  Do we have an input field on this from the new UI Client/Admission screen?  The new system will intake DACODS surveys and attach them to our internal Patient.id.  Confirm if we need to input this value.  Delete it if we don't need it to input it.

KasiSribal replied:
We do not have any input field on DAcodsid so removing it

enjoysparkling replied:
Thank you for checking!

enjoysparkling reviewed (changes requested):
On all the slices we need CARDINALITY 1..1 for identfier:SLICE.system and identfier:SLICE.value

KasiSribal commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -44,6 +44,216 @@
+      {
+        "id": "Patient.identifier:MedicaidID.type.text",
+        "path": "Patient.identifier.type.text",
+        "definition": "A human language representation of the concept as seen/selected/uttered by the user who entered the data and/or which represents the intended meaning of the user.",
```
Removed the definition for Medicaid ID ->type-> text

KasiSribal commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -44,6 +44,216 @@
+        "fixedString": "ClientId"
+      },
+      {
+        "id": "Patient.identifier:ProviderId",
```
Removed ProviderId

KasiSribal commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -44,6 +44,216 @@
+        "fixedString": "Social Security Number"
+      },
+      {
+        "id": "Patient.identifier:SSN.system",
```
Changed CARDINALITY on all of these identifiers to be 0..*.

KasiSribal commented:
Changes:
1) Changed CARDINALITY on all of these identifiers to be 0..*. 
2) Removed the definition for Medicaid ID ->type-> text 
3) Removed ProviderId 
4) Removed System -->Fixed value and added definition for ProviderClientId 
5) Removed type.coding.system, for Patient.identifier:ClientId which represents Signal's Beacon generated number 
6) Client id label to "Signal's Beacon legacy identifier for client." 
7) assigned 'Patient.identifier:ClientID.system' as 
https://signalbhn.org/fhir/sid/beacon-legacy
 8) We do not have any input field on DAcodsid so removing it 
9) On all the slices assigned cardinality 1..1 for identfier:SLICE.system and identfier:SLICE.value

enjoysparkling commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -45,6 +45,216 @@
+      {
+        "id": "Patient.identifier:SSN.type.coding.display",
+        "path": "Patient.identifier.type.coding.display",
+        "fixedString": "SSN"
```
I was reading through https://www.hl7.org/fhir/valueset-identifier-type.html, change SSN to be this:
type.coding.display = Social Beneficiary Identifier

KasiSribal replied:
changed

enjoysparkling commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -45,6 +45,216 @@
+        "fixedUri": "https://www.hl7.org/fhir/valueset-identifier-type.html"
+      },
+      {
+        "id": "Patient.identifier:MedicaidID.type.coding.code",
```
Was reading this https://jira.hl7.org/browse/FHIR-27538, chaange MedicaidID and Medicare ID:
type.coding.code = SB
type.coding.display = Social Beneficiary Identifier

KasiSribal replied:
Changed

enjoysparkling commented on input/resources/StructureDefinition-profile-patient.json:
```diff
@@ -45,6 +45,216 @@
+      {
+        "id": "Patient.identifier:MedicareID.type.coding.code",
+        "path": "Patient.identifier.type.coding.code",
+        "fixedCode": "MR"
```
Was reading this https://jira.hl7.org/browse/FHIR-27538, chaange MedicaidID and Medicare ID:
type.coding.code = SB
type.coding.display = Social Beneficiary Identifier

KasiSribal replied:
changed

enjoysparkling reviewed (approved):
Looks good to go!  Thank you, @KasiSribal.
